packages/api/models/rules.js

```diff
@@ -206,67 +204,38 @@ class Rule extends Manager {
    */
   async addKinesisEventSource(item, lambda) {
     // use the existing event source mapping if it already exists and is enabled
-    const listParams = { FunctionName: lambda.name };
-    const listData = await aws.lambda(listParams).listEventSourceMappings().promise();
+    const listParams = {
+      FunctionName: lambda.name,
+      EventSourceArn: item.rule.value
+    };
+    const listData = await aws.lambda().listEventSourceMappings(listParams).promise();
     if (listData.EventSourceMappings && listData.EventSourceMappings.length > 0) {
-      const mappingExists = listData.EventSourceMappings
-        .find((mapping) => { // eslint-disable-line arrow-body-style
-          return (mapping.EventSourceArn === item.rule.value
-                  && mapping.FunctionArn.includes(lambda.name));
-        });
-      if (mappingExists) {
-        if (mappingExists.State === 'Enabled') {
-          return mappingExists;
-        }
-        await this.deleteKinesisEventSource({
-          name: item.name,
-          rule: {
-            arn: mappingExists.UUID,
-            type: item.rule.type
-          }
-        }, lambda.eventType);
+      const mappingExists = listData.EventSourceMappings[0];
+      if (mappingExists.State === 'Enabled') {
+        return mappingExists;
       }
+      await this.deleteKinesisEventSource({
+        name: item.name,
+        rule: {
+          arn: mappingExists.UUID,
+          // This is needed for the query to determine if the event source is shared
+          logEventArn: mappingExists.UUID,
+          type: item.rule.type
+        }
+      }, lambda.eventType);
     }
 
     // create event source mapping
     const params = {
       EventSourceArn: item.rule.value,
       FunctionName: lambda.name,
       StartingPosition: 'TRIM_HORIZON',
-      Enabled: item.state === 'ENABLED'
+      Enabled: true
     };
     const data = await aws.lambda().createEventSourceMapping(params).promise();
```

Comment:
You could just return this Promise instead of `await`ing it.

Reply:
done.